nautobot/virtualization/filters.py

```diff
@@ -233,8 +303,23 @@ class VMInterfaceFilterSet(BaseFilterSet, StatusModelFilterSetMixin, CustomField
     mac_address = MultiValueMACAddressFilter(
         label="MAC address",
     )
-    tag = TagFilter()
+    tagged_vlans = NaturalKeyOrPKMultipleChoiceFilter(
+        natural_key="vid",
+        queryset=VLAN.objects.all(),
+        label="Tagged VLAN (VID or ID)",
+    )
+    untagged_vlan = NaturalKeyOrPKMultipleChoiceFilter(
+        natural_key="vid",
+        queryset=VLAN.objects.all(),
+        label="Untagged VLAN (VID or ID)",
+    )
+    ip_addresses = MultiValueCharFilter(method="filter_ip_address", label="IP Addresses")
```

Comment:
I'm finding myself questioning the `MultiValueCharFilter` here. Means we can't use IPAddress PKs on this filter, only CIDR strings. I wonder if instead we should create a bespoke subclass of `NaturalKeyOrPKMultipleChoiceFilter` that understands how to decompose the CIDR string into a `host, prefix_len` tuple and use that for filtering, but also still supports filtering by PK? Maybe that should be a followup issue though?

Reply:
True, I could work on a bespoke subclass on a different issue, for now, I revamp the `IPAddressQuerySet.filter_address()` to accommodate filtering on `address` and/or `pk`.